src/sardana/macroserver/msmacromanager.py

```diff
@@ -74,7 +77,32 @@
 
 _BASE_DIR = os.path.dirname(os.path.abspath(__file__))
 
-
+class LogFilter(logging.Filter):
+    # Possible fields: created, name, process, levelname, levelno, funcName, module, message = msg
+    # Ex.:
+    # created   -> 1515147905.34
+    # name      -> Door/issue102/1.Macro[ascan]
+    # process   -> 15638
+    # levelname -> OUTPUT
+    # levelno   -> 15
+    # funcName  -> output
+    # module    -> __init__
+    # message   -> 2         10        0         0         0         0      2.11293 
+
+    def __init__(self, param=None, field=None):
+        self.param = param
+        self.field = field
+
+    def filter(self, record):
+        allow = True
+        if self.param != "MacroExecutor" and "MacroExecutor" in record.name:
```

Comment:
It took me some time to understand the logic of `LogFilter`. What do you think to refactor it to make it more intuitive. If I remember correctly, this precise condition is for filtering out some messages in case of executing nested macros. Maybe adding an extra argument to the constructor, let's call it `logger` and make the matching with the regular expression could make it easier to follow.

Reply:
Hi @reszelaz, sorry, I do not understand which is the change that you propose. It is using a matching with regular expresions ... the line you mention is just an exception for keeping the line in which the macro name and arguments appear.

Reply:
Sorry for the late reply! I think it would be easier if I propose a change directly instead of explaining it. Then you will evaluate it. Soon I will make a PR to your branch.